### PR TITLE
Postgres connect with url string

### PR DESCRIPTION
**Pull request by borsemayur2**

### CODE REVIEW

**Comment by borsemayur2** [comment]
Related to #253

File: lib/connectors/postgres-connector.ts [hunk 1]
```diff
@@ -5,14 +5,28 @@ import type { SupportedSQLDatabaseDialect } from "../translators/sql-translator.
...
 
+interface PostgresOptionsWithURI extends ConnectorOptions {
+  uri: string;
+  // unused types
+  database?: never;
+  host?: never;
+  username?: never;
+  password?: never;
+  port?: never;
+}
```
Comment (eveningkid) [inline lib/connectors/postgres-connector.ts:26]: This notation really confused me. Shouldn't it be:

```ts
interface PostgresOptionsWithConfig extends ConnectorOptions {
  database: string;
  host: string;
  username: string;
  password: string;
  port?: number;
}

interface PostgresOptionsWithURI extends ConnectorOptions {
  uri: string;
}

export type PostgresOptions = PostgresOptionsWithConfig | PostgresOptionsWithURI;
```

**Review by eveningkid — CHANGES_REQUESTED** [review]
Hey Mayur,

Thank you for contributing to the project :)

I left you a note, I'll wait for your reply.

Have a great day

File: lib/connectors/postgres-connector.ts (hunk 1 above)
Comment (borsemayur2) [reply lib/connectors/postgres-connector.ts:26]: This was indeed my initial approach. But TypeScript throws error so I used `never`.
image
image

Comment (eveningkid) [reply lib/connectors/postgres-connector.ts:26]: Alright, the reason is `hasOwnProperty` doesn't [work as a typescript type guard](https://stackoverflow.com/a/59200046).

So you can keep your original thought and replace the check with:

```js
if ("uri" in options) {
  // shouldn't raise an error
  this._client = new PostgresClient(options.uri);
}
```

That's great to know! I would have done the same mistake :D

Comment (borsemayur2) [reply lib/connectors/postgres-connector.ts]: done

**Review by eveningkid — APPROVED** [review]

**Review by eveningkid — APPROVED** [review]
All good, Mayur :)

Thanks, will merge now and make a new release!

Have a great day

**Pull request merged**